xbmc/games/addons/input/GameClientInput.cpp

```diff
@@ -288,6 +288,95 @@ bool CGameClientInput::SupportsMouse() const
   return it != controllers.Ports().end() && !it->CompatibleControllers().empty();
 }
 
+bool CGameClientInput::ConnectController(const std::string& portAddress, ControllerPtr controller)
+{
+  const CControllerTree& controllerTree = m_topology->ControllerTree();
+
+  // Validate controller
+  const CPortNode& port = controllerTree.GetPort(portAddress);
+  if (!port.IsControllerAccepted(portAddress, controller->ID()))
+  {
+    CLog::Log(LOGERROR, "Failed to open port: Invalid controller \"{}\" on port \"{}\"",
+              controller->ID(), portAddress);
+    return false;
+  }
+
+  // Close current ports if any are open
+  const CPortNode& currentPort = m_topology->ControllerTree().GetPort(portAddress);
+  CloseJoysticks(currentPort);
+
+  bool bSuccess = false;
+
+  {
+    CSingleLock lock(m_clientAccess);
+
+    if (m_gameClient.Initialized())
```

Comment:
Return early if `m_gameClient.Initialized()` is `false`? And return early from the `catch` scope? Same below in `CGameClientInput::DisconnectController()`.

Reply:
Fixed